Realm/Realm/Sync/App.cs

```diff
@@ -150,6 +153,8 @@ public static App Create(AppConfiguration config)
                 }
             }
 
+            var httpClient = new HttpClient(config.HttpClientHandler ?? new HttpClientHandler());
```

Comment:
I think we should initialize the `AppConfiguration.HttpClientHandler` property with the default value, instead of leaving it null and instantiating the default handler here.